### PR TITLE
hotfix: address instead of username in chat messages

### PR DESCRIPTION
**Pull request by lorux0**
## What does this PR change?

Fetches the profile, if not available, when adding a chat message.
Fixes that sometimes you see the address of the user instead of the name in the message, either sender or recipient. 

## How to test the changes?

1. Launch the explorer
2. Send and receive messages in ~nearby
3. Send and receive messages in private conversations
4. Send and receive messages in #channels (recommended to use #test)
5. If at any point you see an address instead of the user name, it should change to a proper name after a small period of time

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by sandrade-dcl — APPROVED** [review]
LGTM, although I see that the code flow of `EnsureProfileThenUpdateMessage` and its retries approach is a bit complex and could be also a bit delicated.

Is it really needed to implement a retries system here?, why could be needed to call `RequestFullUserProfileAsync` several times?, because of possible connection issues for example?.

We're going to have to do a deep test of this feature 👍

**Comment by lorux0** [comment]
> LGTM, although I see that the code flow of `EnsureProfileThenUpdateMessage` and its retries approach is a bit complex and could be also a bit delicated.
> 
> Is it really needed to implement a retries system here?, why could be needed to call `RequestFullUserProfileAsync` several times?, because of possible connection issues for example?.
> 
> We're going to have to do a deep test of this feature 👍

I added the retries system because there is a recursive call to `AddChatMessage` in `EnsureProfileThenUpdateMessage` which may freeze the app if not handled correctly

**Review by sandrade-dcl — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs [hunk 1]
```diff
@@ -109,29 +118,131 @@ public void SetVisibility(bool visible)
...
+            {
+                Debug.LogWarning($"Profile fetching max retries limit reached for {profileId}");
+                return;
+            }
+
+            try
+            {
+                UserProfile requestedProfile = await userProfileBridge.RequestFullUserProfileAsync(profileId, cancellationToken);
+
+                if (requestedProfile != null)
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs:139]: Just a highlight: according to the logic underneath `userProfile` can never be `null`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs [hunk 2]
```diff
@@ -109,29 +118,131 @@ public void SetVisibility(bool visible)
...
+                : ChatEntryModel.SubType.RECEIVED;
+        }
+        else if (message.messageType == ChatMessage.Type.PUBLIC)
+        {
+            model.subType = message.sender == ownProfile.userId
+                ? ChatEntryModel.SubType.SENT
+                : ChatEntryModel.SubType.RECEIVED;
+        }
+
+        AddChatMessage(model, setScrollPositionToBottom, spamFiltering, limitMaxEntries,
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs:222]: This recursive flow is hard to follow. I would like to clarify how it should work:
1. It is called `AddChatMessage` but when called multiple times it acts as `update`?
2. Why do we call right the same method from `EnsureProfileThenUpdateMessage` when we already have `model` created and the only thing we need to do is update `recipientName` ? It leads to additional allocations and logical misintentions.

**Review by mikhail-dcl — COMMENTED** [review]
I am not going to block the hotfix but I think the code could be written in a neater way

**Comment by lorux0** [comment]
Base branch changed to `dev`. Check this pr: https://github.com/decentraland/unity-renderer/pull/4800
